tests/test_tokenization_common.py

```diff
@@ -1153,6 +1153,122 @@ def test_chat_template_batched(self):
+                {"role": "user", "content": "user message 2"},
+                {"role": "assistant", "content": "<start2> assistant message 6"},
+            ],
+            [
+                {"role": "system", "content": "system message 3"},
+                {"role": "user", "content": "user message 3"},
+                {"role": "assistant", "content": "<start3> assistant message 3 7"},
+                {"role": "user", "content": "user message 4"},
+                {"role": "assistant", "content": "<start4> assistant message 4 8"},
+            ],
+        ]
+        assistant_wrappers = [
+            [("<start>", "5<|im_end|>"), ("<start2>", "6<|im_end|>")],
+            [("<start3>", "7<|im_end|>"), ("<start4>", "8<|im_end|>")],
+        ]
+        for tokenizer, pretrained_name, _ in self.tokenizers_list:
+            with self.subTest(f"{tokenizer.__class__.__name__} ({pretrained_name})"):
+                if not self.test_rust_tokenizer:
+                    self.skipTest(reason="No fast tokenizer defined")
+
+                tokenizer_r = self.rust_tokenizer_class.from_pretrained(pretrained_name)
+
+                # check batched
+                output = tokenizer_r.apply_chat_template(
+                    conversations,
+                    chat_template=dummy_template,
+                    tokenize=True,
+                    return_assistant_tokens_mask=True,
+                    return_dict=True,
+                )
+                for i, conv in enumerate(conversations):
+                    chat_string = tokenizer_r.apply_chat_template(
+                        conversations[i], tokenize=False, chat_template=dummy_template
+                    )
+                    assistant_start = output.char_to_token(i, chat_string.index(assistant_wrappers[i][0][0]))
+                    assistant_end = output.char_to_token(
+                        i, chat_string.index(assistant_wrappers[i][0][1]) + len(assistant_wrappers[i][0][1]) - 1
+                    )
+
+                    assistant_start2 = output.char_to_token(i, chat_string.index(assistant_wrappers[i][1][0]))
+                    assistant_end2 = output.char_to_token(
+                        i, chat_string.index(assistant_wrappers[i][1][1]) + len(assistant_wrappers[i][1][1]) - 1
+                    )
+                    # assert labels in assistant indices
+                    self.assertEqual(
+                        output["assistant_masks"][i][assistant_start : assistant_end + 1],
+                        [1] * (assistant_end - assistant_start + 1),
+                    )
+                    self.assertEqual(
+                        output["assistant_masks"][i][assistant_start2 : assistant_end2 + 1],
+                        [1] * (assistant_end2 - assistant_start2 + 1),
+                    )
+                    # assert 0 in user/system indices
+                    self.assertEqual(all(l == 0 for l in output["assistant_masks"][i][:assistant_start]), True)
```

Comment:
It would be better to check the logic structure consistent here with the checks above e.g. testing `all` versus checking a list constructed of the expected length of 0s or 1s. FWIW, I think the latter is better as although slower will be easier to debug if tests start failing